src/hardware/adc.rs

```diff
@@ -29,44 +51,66 @@ pub enum InputChannel {
     Seven = 7,
 }
 
+// Physical ADC devices on Thermostat
+#[derive(Clone, Copy)]
+
+pub enum AdcPhy {
+    Zero = 0,
+    One = 1,
+    Two = 2,
+    Three = 3,
+}
+
 type O = Output<PushPull>;
-type Adcs = (
-    ad7172::Ad7172<SharedBus<Spi<SPI4, Enabled>>, PE0<O>>,
-    ad7172::Ad7172<SharedBus<Spi<SPI4, Enabled>>, PE1<O>>,
-    ad7172::Ad7172<SharedBus<Spi<SPI4, Enabled>>, PE3<O>>,
-    ad7172::Ad7172<SharedBus<Spi<SPI4, Enabled>>, PE4<O>>,
-);
+type Adcs = ad7172::Ad7172<Spi<SPI4, Enabled>>;
 type SpiPins = (
     PE2<Alternate<AF5>>,
     PE5<Alternate<AF5>>,
     PE6<Alternate<AF5>>,
 );
-
+/// All pins for all ADCs.
+/// * `spi` - Spi clk, miso, mosi (in this order).
```

Comment:
I would just `allow(complex_type)` or whatever clippy is complainitng about. The cryptic `O` type alias and separate `SpiPins` (where the tuple index doesn't have semantic value) don't help with readability here.